src/main/java/org/null0/edi/debatcher/Extractor1.java

```diff
@@ -1,598 +0,0 @@
-			edifileproperties.load(in);
-			fileproperties = new FileProperties();
-			// Set the file delimiters which will be used for splitting the lines
-			fileproperties.setDataElementSeparator(edifileproperties.getProperty("dataElementSeperator").charAt(0));
-			fileproperties.setSegmentTerminator(edifileproperties.getProperty("segmentTerminator").charAt(0));
-			fileproperties.setISAIEAFolderName(edifileproperties.getProperty("isaieafoldername"));
-			fileproperties.setGSGEFolderName(edifileproperties.getProperty("gsgefoldername"));
-			fileproperties.setSTSEFolderName(edifileproperties.getProperty("stsefoldername"));
-			fileproperties.setParentFolderName(edifileproperties.getProperty("rootfolderpath"));
-			fileproperties.setOverrideFileSize(Boolean.parseBoolean(edifileproperties.getProperty("overridefilesize")));
-			Log.debug("Finished the properties file..");
-		} catch (Exception e) {
-			Log.error("IO Error in ReadProperties", e.getMessage());
-			throw e;
-		}
-	}
-
-	// Delete the temp. directories that were created; we don't need those any more
-	private boolean DeleteTempDirectories(boolean delete) {
-		if (delete) {
-			File tempdirectory = new File(isafolder.getPath());
-			return tempdirectory.exists() ? tempdirectory.delete() : false;
-		} else {
-			return false;
-		}
-	}
-
-	private boolean CreateFolders(String rootfoldername, String currentdatetime) {
-		return CreateDateTimeFolder(rootfoldername, currentdatetime);
-	}
-
-	private boolean CreateDateTimeFolder(String rootfoldername, String currentdatetime) {
-		datetimefolder = new File(rootfoldername + currentdatetime);
-		return !datetimefolder.exists() ? datetimefolder.mkdir() : false;
-	}
-
-	private boolean CreateInterchangeControlFolder(String rootfolder, String isaisefolderpath) {
-		isafolder = new File(rootfolder + "/" + isaisefolderpath);
-		return !isafolder.exists() ? isafolder.mkdir() : false;
-	}
-
-	private boolean CreateFunctionalGroupFolders(String isaieafolderpath, String gsgefolderpath) {
-		gsfolder = new File(isaieafolderpath + "/" + gsgefolderpath);
-		return !gsfolder.exists() ? gsfolder.mkdir() : false;
-	}
-
-	private boolean CreateTransactionalSetFolders(String gsgefolderpath, String stsefolderpath) {
-		stfolder = new File(gsgefolderpath + "/" + stsefolderpath);
-		return !stfolder.exists() ? stfolder.mkdir() : false;
-	}
-
-	private boolean CreateFinalOutputFolders(String rootfolder) {
-		clmfolder = new File(rootfolder + "/" + "Claims");
-		return !clmfolder.exists() ? clmfolder.mkdir() : false;
-	}
-
-	public void setFileSize(long filesize) {
-		this.filesize = filesize;
-	}
-
-	public long getFileSize() {
-		return this.filesize;
-	}
-
-	public String CurrentDateTime() {
-		SimpleDateFormat dateformat = new SimpleDateFormat("yyyy-MM-dd_HH:mm:ss:SSS");
-		Date currentdate = new Date();
-		this.currentdatetime = dateformat.format(currentdate);
-		return this.currentdatetime;
-	}
-
-	public boolean Debatch() throws Exception {
```

Comment:
Alternative attempt to write a debatcher. Implementation did not succeed correctly. Agree on the fact this should be removed. Did this ever work?

Reply:
I'm not sure, don't think it was finished. I believe it may have worked on unwrapping the first layers of envelopes, just the intermediate files being written out. I don't think the final output was generated or tested. There was a big TODO to handle large files, and when I saw all the overhead to write and re-read the files I proposed an alternative approach that would scale.